### PR TITLE
HYDRA-776 : Update build doc & script

### PR DESCRIPTION
**Pull request by debloip-adsk**
This PR updates the build docs to (hopefully) contain all info that might be needed to build the plugin. It also contains a slight tweak to the top-level CMakeLists to automatically determine the default Python version to build with based on the Maya version : note that the Python version can still be manually specified with the same variable as before.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -106,8 +112,6 @@ message(STATUS "   PYTHON_INCLUDE_DIRS = ${PYTHON_INCLUDE_DIRS}")
 message(STATUS "   PYTHON_LIBRARIES    = ${PYTHON_LIBRARIES}")
 message(STATUS "   Python_EXECUTABLE   = ${Python_EXECUTABLE}")
 
-include(cmake/jinja.cmake)
```
Comment (debloip-adsk) [inline CMakeLists.txt:109]: Jinja is no longer needed in MayaHydra

File: doc/build.md [hunk 2]
```diff
@@ -135,9 +159,9 @@ Examples:
 
 It is up to the user to select the CMake Generator of choice, but we encourage the use of the Ninja generator. To use the Ninja Generator, you need to first install the Ninja binary from https://ninja-build.org/
 
-You then need to set the generator to ```Ninja``` and the ```CMAKE_MAKE_PROGRAM``` variable to the Ninja binary you downloaded.
+You then need to set the generator to ```Ninja```.
 ```
-python build.py --generator Ninja --build-args=-DCMAKE_MAKE_PROGRAM='path to ninja binary'
+python build.py --generator Ninja
```
Comment (debloip-adsk) [inline doc/build.md:164]: Specifying the CMAKE_MAKE_PROGRAM should not be necessary as CMake will figure out which one to use based on the generator. This option mostly seems to be used to force a specific "make" program.

File: doc/build.md [hunk 3]
```diff
@@ -183,8 +207,6 @@ e.g
 ➜ pip list
 Package    Version
 ---------- --------
-Jinja2     3.1.2
-MarkupSafe 2.1.1
```
Comment (debloip-adsk) [inline doc/build.md:187]: These two are no longer relevant

**Review by lilike-adsk — APPROVED** [review]

**Comment by debloip-adsk** [comment]
FYI @lanierd-adsk @roopavr-adsk @ppt-adsk

**Pull request merged**